src/scripts/activate-user.ts

```diff
@@ -21,15 +36,56 @@ export const activateUser = (
         O.fold(
           () => TE.left(`User [${userId}] does not exist. Cannot activate.`),
           // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
-          () => service.activateUser(userId)
+          (user) => TE.right(user)
         )
       )
-    ),
-    TE.chainFirstIOK((user) =>
-      Console.info(`Activated [${user.id}] [${user.email}].`)
     )
   );
 
+export const checkUserStatusPriorToActivation = (
+  user: User
+): TE.TaskEither<string, User> => {
```

Comment:
**thought:** notice that the input type (`User`) and the return type (`User`) are the same. This normally means that we are *validating*, not *parsing*.

In the future, consider whether it is worth while "narrowing" the return type to something like `ActivatableUser` and requiring that `activateUser` takes an `ActivatableUser`, not a `User`.